lib/registry/src/types.rs

```diff
@@ -24,3 +26,41 @@ pub struct PublishDeployAppOutput {
     pub version_name: String,
     pub owner_name: String,
 }
+
+#[derive(Clone, Debug)]
+pub struct NewNonceOutput {
+    pub auth_url: String,
+}
+
+/// Payload from the frontend after the user has authenticated.
+#[derive(Clone, Debug, Deserialize)]
+#[serde(rename_all = "lowercase")]
+pub enum TokenStatus {
+    Cancelled,
+    Authorized,
+}
+
+fn token_status_deserializer<'de, D>(deserializer: D) -> Result<TokenStatus, D::Error>
+where
+    D: serde::Deserializer<'de>,
+{
+    let s = String::deserialize(deserializer)?;
+    match s.as_str() {
+        "cancelled" => Ok(TokenStatus::Cancelled),
+        "authorized" => Ok(TokenStatus::Authorized),
+        _ => Err(serde::de::Error::custom(format!(
+            "invalid token status: {}",
+            s
+        ))),
+    }
+}
+
+/// Payload from the frontend after the user has authenticated.
+///
+/// This has the token that we need to set in the WASMER_TOML file.
+#[derive(Clone, Debug, Deserialize)]
+pub struct ValidatedNonceOutput {
+    pub token: String,
+    #[serde(deserialize_with = "token_status_deserializer")]
+    pub status: TokenStatus,
+}
```

Comment:
The `TokenStatus` should not live in the registry, since it's tied to the HTTP auth server. We should move it to `lib/cli/src/commands/login.rs`

Reply:
@syrusakbary ValidatedNonceOutput also depends TokenStatus. Which is the returned response from graphQL but that too is used only in HTTP auth server, Should I also move that to `wasmer-cli`.

Otherwise, I'm running into the problem of cyclic dependencies. 
As `wasmer-cli` depends on `wasmer-registry` but in the case you specified `wasmer-cli` only has the `TokenStatus` type which will exist in `wasmer-cli` but `wasmer-registry` uses it in `ValidatedNonceOutput`.